crates/ty_python_semantic/src/program.rs

```diff
@@ -163,15 +163,195 @@ impl Default for PythonVersionWithSource {
     }
 }
 
-#[derive(PartialEq, Eq, Debug, Copy, Clone, get_size2::GetSize)]
-pub enum MisconfigurationMode {
-    /// Settings Failure Is Not An Error.
+/// A type that never exists.
+///
+/// In Rust if you have Result<T, Never> the compiler knows `Err` is impossible
+/// and you can just write `let Ok(val) = result;`
+pub enum Never {}
```

Comment:
Could we make use of this in many places where we use `.unwrap` today?

Reply:
I think you can just use `std::convert::Infallible` here instead. (I don't know if it really matters. Probably not. But seems nicer to use the same type as std.)